CondCore/EcalPlugins/plugins/EcalChannelStatus_PayloadInspector.cc

```diff
@@ -0,0 +1,101 @@
+#include "CondCore/Utilities/interface/PayloadInspectorModule.h"
+#include "CondCore/Utilities/interface/PayloadInspector.h"
+#include "CondCore/CondDB/interface/Time.h"
+
+// the data format of the condition to be inspected
+#include "CondFormats/EcalObjects/interface/EcalChannelStatus.h"
+
+#include <memory>
+#include <sstream>
+
+
+
+namespace {
+
+  /*******************************************************
+   
+     2d histogram of ECAL barrel channel status of 1 IOV 
+
+  *******************************************************/
+
+
+  // inherit from one of the predefined plot class: Histogram2D
+  class EcalChannelStatusEBMap : public cond::payloadInspector::Histogram2D<EcalChannelStatus> {
+
+  public:
+    EcalChannelStatusEBMap() : cond::payloadInspector::Histogram2D<EcalChannelStatus>( "ECAL Barrel channel status - map ",
+										       "iphi", 360, 1., 361., "ieta", 171, -85., 86.) {
+     Base::setSingleIov( true );
+    }
+
+    // Histogram2D::fill (virtual) needs be overridden - the implementation should use fillWithValue
+    bool fill( const std::vector<std::tuple<cond::Time_t,cond::Hash> >& iovs ){
+
+      for( auto iov : iovs ) {
+	std::shared_ptr<EcalChannelStatus> payload = Base::fetchPayload( std::get<1>(iov) );
+	if( payload.get() ){
+	  // looping over the EB channels, via the dense-index, mapped into EBDetId's
+	  if (!payload->barrelItems().size()) return false;
+	  for(int cellid = EBDetId::MIN_HASH;
+	      cellid < EBDetId::kSizeForDenseIndexing;
+	      ++cellid) {
+	    uint32_t rawid = EBDetId::unhashIndex(cellid);
+	    
+	    // check the existence of ECAL channel status, for a given ECAL barrel channel
+	    if (payload->find(rawid) == payload->end()) continue;
+	    if (!(*payload)[rawid].getEncodedStatusCode()) continue;
+
+	    // fill the Histogram2D here
+	    fillWithValue(  (EBDetId(rawid)).iphi() , (EBDetId(rawid)).ieta(), (*payload)[rawid].getEncodedStatusCode());
+	  }// loop over cellid
+	}// if payload.get()
+      }// loop over IOV's (1 in this case)
+
+      return true;
+    }// fill method
+  };
+
+  class EcalChannelStatusEEMap : public cond::payloadInspector::Histogram2D<EcalChannelStatus> {
+
+  public:
+    EcalChannelStatusEEMap() : cond::payloadInspector::Histogram2D<EcalChannelStatus>( "ECAL Barrel channel status - map ",
```

Comment:
@depasse I think this is a copy paste error. It should be ECAL Endcap channel status. Can you please confirm and eventually change if its true?